R/authors.R

```diff
@@ -0,0 +1,103 @@
+#' Create or modify the `Authors@R` field in the DESCRIPTION file
+#'
+#' @description
+#' If the `Authors@R` field is not included in the DESCRIPTION file then `use_author()` adds the field assigns
+#' the author defined by the input parameters. If the `Authors@R` field exists already in the DESCRIPTION then
+#' `use_author()` will add the author defined by the input parameters as an additional new author.
+#'
+#' The `use_author()` function should be used after [create_package()] or [use_description()].
+#'
+#' If you create a lot of packages, consider storing personalized defaults as a
+#' named list in an option named `"usethis.description"`. [use_description()] will automatically
+#' fill using this information. Here's an example of code to include in `.Rprofile`:
+#'
+#' ```
+#' options(
+#'   usethis.description = list(
+#'     `Authors@R` = 'person("Jane", "Doe", email = "jane@example.com", role = c("aut", "cre"),
+#'                           comment = c(ORCID = "YOUR-ORCID-ID"))',
+#'     License = "MIT + file LICENSE",
+#'     Language =  "es"
+#'   )
+#' )
+#' ```
+#'
+#' @param given a character string with the given (first) name of the author.
+#' @param family a character string with the family (last) name of the author.
+#' @param role a character vector specifying the role of the person to be added.
+#' @param email a character string giving an e-mail address of the author to be added.
+#' @param comment a character string providing comments related to the author to be added.
+#'
+#' @export
+#'
+#' @importFrom desc desc_add_author desc_fields desc_get_authors desc_set_authors
+#' @importFrom utils person
+#'
+#' @examples
+#' \dontrun{
+#' # Adds the default Jane Doe to author
+#' use_author()
+#'
+#' # Adds the author information for Ali
+#' use_author(given = "Ali",
+#'           family = "Val",
+#'           role = c("aut", "cre"),
+#'           email = "alval@example.com",
+#'           comment = c(ORCID = "YOUR-ORCID-ID"))
+#'
+#' # Adds a second author Ali2
+#'use_author(given = "Ali2",
+#'           family = "Val2",
+#'           role = c("cph", "fnd"),
+#'           email = "alval2@example.com",
+#'           comment = NULL)
+#' }
+#'
+
+use_author <- function(given = "Jane", family = "Doe", role = c("aut", "cre"), email = "jane@example.com", comment = c(ORCID = "YOUR-ORCID-ID")){
+  # Adapted from use_dependency code and tools provided in the desc package
+  # TODO allow generic authors input like desc_set_authors(authors, file = ".", normalize = FALSE) for multiple author input
+  # TODO create addin to prompt for author information
+  # TODO create a snippet with author information for DESCRIPTION
+  # TODO add tests
+
```

Comment:
There should be some checks for obviously bad input here. The internal helper `is_string()` will be useful to you.

Reply:
I didn't manually check inputs since the inputs for my function are all passed to desc package functions. They do seems to have checks on inputs when someone is input incorrectly as a vector or numeric for example. I'm happy to add these checks if you prefer they be done within the usethis function directly.